### ISSUE TITLE
Model performance on test set does not match reported results

### ISSUE THREAD

**Issue by cheng1559** [qiyan98/TrajFlow#3]
Hi authors,

Thank you for sharing your work! I was attempted to reproduce the test set result following by the `README` steps. However, I noticed there is a significant gap between my reults and the ones in the paper. Below is the comparison:

Metric | Paper | Reproduced
-- | -- | --
Soft mAP v2 | 0.4710 | 0.4572
mAP v2 | 0.4604 | 0.4466
Min ADE | 0.5714 | 0.5811
Min FDE | 1.1667 | 1.1846
Miss Rate | 0.1162 | 0.1193
Overlap Rate | 0.1272 | 0.1296

I would like to ask if you have any suggestions about what might be causing this discrepancy.

Also, I noticed that the paper mentions training on `8` GPUs, but the training script you provided specifies only `4` GPUs. Could this difference in the number of GPUs have an impact on the test performance (although the same total batch size is the same)?

Looking forward to your reply, and thanks again for your excellent work!

**Comment by qiyan98** [comment]
Hi there,

Thanks for your interest in our work. During the camera-ready preparation stage, we reproduced the experiments on different clusters (A40s/A100s/V100s) with various configurations and found that checkpoint performance did vary a bit, especially when the batch size was quite large.

We initially used 4× A40 48GB GPUs with a batch size of 10 per GPU. We then re-ran the experiments on 8× A100 80GB GPUs with the same per-GPU batch size. Both experiments produced similar performance, and the results were consistent with the numbers reported in the table (which we also used for the leaderboard submission). Later, we experimented with larger per-GPU batch sizes on A100s and V100s, and found that late-stage checkpoint performance was not as good—consistent with what you observed in your experiments. We conjecture that the optimization process may have played a ‘weird’ role here.

Unfortunately, we no longer have as much access to A100 GPUs as before and are unable to dedicate further resources to investigate this phenomenon. We hope this context clarifies the variability and provides some useful tips for your own experiments.

Best,
Qi

**Comment by leandro-svg** [comment]
Dear, 

Following the discussion by @cheng1559, I have attempted to reproduce your results on both the validation and test sets using the provided config file and similar hardware specifications (8 A100 80GB GPUs, 10 batches/GPU). However, I was only able to achieve approximately 0.44 mAP:

text
******************** denoiser output results ********************
       Waymo          mAP       minADE       minFDE     MissRate            
      VEHICLE      0.4665,      0.6873,      1.4078,      0.1190,            
   PEDESTRIAN      0.4637,      0.3451,      0.7146,      0.0663,            
      CYCLIST      0.3852,      0.6965,      1.4391,      0.1707,            
          Avg      0.4385,      0.5763,      1.1872,      0.1187,  

I was wondering if there might be a discrepancy between the uploaded config file in the repository and the one used to generate the reported results, or if there are any additional training details I may have missed.

I would greatly appreciate any guidance you could provide.
